src/components/Checkout/Checkout.component.jsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { useSelector } from 'react-redux';
+
+import { checkoutActionCreators } from 'state/index';
+import { bindActionCreators } from 'redux';
+import { useDispatch } from 'react-redux';
+import { useHistory } from "react-router-dom";
+
+
+//checkout data:
+// const checkoutDataModel = {
+//     orderSummary: [{ name, quantity, subtotal }, ...]
+//     total
+// };
+
+const CheckoutComponent = ({ className }) => {
+
+    const { orderSummary, total } = useSelector((state) => state.checkout);
+    const dispatch = useDispatch()
+    const { clearCheckoutSummary } = bindActionCreators(checkoutActionCreators, dispatch);
+
+    const history = useHistory();
+
+    const handleOnSubmit = () => {
+        history.push("/home");
+    };
+
+
+  return (
+    <form onSubmit={handleOnSubmit}>
+        <div className={className}>
+        
+            <div className="checkout-customerdata">
+                <label htmlFor="name">Customer Name:</label>
+                <input name="name" type="text"></input>
+                <label htmlFor="email">E-mail</label>
+                <input name="email" type="text"></input>
+                <label htmlFor="zipCode">Zip Code</label>
+                <input name="zipCode" type="text"></input>
+                <label htmlFor="orderNotes">Order Notes</label>
+                <textarea name="orderNotes" rows="4" cols="50"></textarea>
+            </div>
+
+            <table>
+            <thead>
+                <tr>
+                <td>Product Name</td>
+                <td>Quantity</td>
+                <td>Subtotal</td>
+                </tr>
+            </thead>
+            <tbody>
+                {
+                    orderSummary.map(({ name, quantity, subtotal}) => 
+                            <tr>
+                                <td>{name}</td>
+                                <td>{quantity}</td>
+                                <td>${subtotal}</td>
+                            </tr>
+                    )
+
+                }
+            </tbody>
+            <tfoot>
+                <tr>
+                <td>Total: ${total}</td>
+                </tr>
+            </tfoot>
+            </table>
+
+            <div className="checkout-actions">
+            <Link to="/cart"><button onClick={clearCheckoutSummary}>Go back</button></Link>
+            <button type="submit">Place Order</button>
+            </div>
+        
+        </div>
+    </form>
+  );
+};
```

Comment:
My suggestion here is to change it to a little bit different structured component. Use the `Checkout.component.jsx` just as a container for the real components that handle something very specific for the page. Try to think this way for everything in the app.
```suggestion
  return (
    // i know classes may need to be changed a little bit to reflect the changes.
    <div className="checkout-container">
       // here goes the form, not all of what you have on the component needs to be part of the form
       <CheckoutForm /> 
       // Then a component for your summary for the checkout
       <CheckoutSummary />
       // Then you can have your action buttons for placing the order or go back 
       <CheckoutAction />
    <div/>
  );
};
```

Reply:
For the checkout form I want to add a deeper suggestion. For forms that are just very repetitive and that you know that have the same structure i'll do something like this. 
Create a simple input component `FormInput.jsx`
```
export const FormInput = ( {name, type, label }) => {
// inside this component then you can even handle the input change and more but for what is required for the 
// deliverable this is fine
   return (
      <div className="input>
         <label htmlFor={name}> {label} </label>
          <input name={name} type={type}></input>
      <div />
   );
}
```

then your checkout form would be

```
export const CheckoutForm = () => {
   // An array that consists of all the necessary inputs
   const checkoutInputs = [ { name:'name', type: 'text', label: 'Customer Name' } ];
   return (
      <>
          {
            checkoutInputs.map( (input) =>  (
                <FormInput {...input} />
            ))
          }
      </>
   );
}
```
For this project there is not a very big form or a lot that you have but believe me when there are projects with forms that have 20, 30 or more inputs and are all the same having this kind of structure helps a lot to maintain and make changes